### PR TITLE
GTEST/UCT: add UCS_OK pending queue test

### PR DESCRIPTION
**Pull request by alex-mikheev**
The test covers DC transports issue the was fixed in
3bb34d4986c11fd68e3204cbd76656d3d900b95c

@yosefe @hoopoepg please take a look

### CODE REVIEW

File: test/gtest/uct/test_pending.cc [hunk 1]
```diff
@@ -328,6 +336,68 @@ UCS_TEST_P(test_uct_pending, pending_async)
...
+    /* set a callback for the uct to invoke when receiving the data */
+    install_handler_sync_or_async(m_e2->iface(), 0, am_handler_simple, 0);
+
+    if (m_e1->iface_attr().cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) {
+        N = 2048;
+    } else {
+        N = 128;
+    }
+
+    if (RUNNING_ON_VALGRIND) {
```
Comment (hoopoepg) [inline test/gtest/uct/test_pending.cc]: what about merge this evaluation and previous one?
like
if (!VALGRIND) {
    if (CONNECTO_TO_IFACE) {
        N = 2048
    } else {
        N = 128
    }
} else {
    N = 64
}

File: test/gtest/uct/test_pending.cc [hunk 2]
```diff
@@ -107,6 +107,14 @@ class test_uct_pending : public uct_test {
         return (ucs_status_t)packed_len;
     }
 
+    static ucs_status_t pending_send_op_ok(uct_pending_req_t *self) {
+        pending_send_request_t *req = ucs_container_of(self, pending_send_request_t, uct);
+
+        delete req;
```
Comment (hoopoepg) [inline test/gtest/uct/test_pending.cc]: it is better practice to have pair of routines alloc/free instead of direct call ```delete```

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6307/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6308/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6318/ for details (Mellanox internal link).

**Comment by alex-mikheev** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6329/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6345/ for details (Mellanox internal link).

**Comment by alex-mikheev** [comment]
bot:retest

**Comment by alex-mikheev** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6347/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6349/ for details (Mellanox internal link).

**Pull request merged**